code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -36,8 +36,8 @@
 	else
 		..()
 
-/obj/item/reagent_containers/food/snacks/proc/On_Consume()
-	if(!usr)
+/obj/item/reagent_containers/food/snacks/proc/On_Consume(mob/M)
+	if(!usr || !M)
```

Comment:
I think you could just replace all instances of usr with M

Reply:
They're different when someone feeds something to someone else.